target/linux/ipq806x/files/arch/arm/boot/dts/qcom-ipq8068-mr42.dts

```diff
@@ -0,0 +1,179 @@
+#include "qcom-ipq8068-cryptid-common.dtsi"
+
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/soc/qcom,tcsr.h>
+
+/ {
+	model = "Meraki MR42";
+	compatible = "meraki,mr42", "qcom,ipq8064";
+
+	aliases {
+		serial1 = &gsbi1_serial;
+		mdio-gpio0 = &mdio_gpio0;
+		ethernet0 = &gmac3;
+
+		led-boot = &led_active_white;
+		led-failsafe = &led_power_orange;
+		led-running = &led_active_white;
+		led-upgrade = &led_active_white;
+	};
+
+	keys {
+		compatible = "gpio-keys";
+		pinctrl-0 = <&button_pins>;
+		pinctrl-names = "default";
+
+		reset {
+			label = "reset";
+			gpios = <&qcom_pinmux 26 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		pinctrl-0 = <&led_pins>;
+		pinctrl-names = "default";
+
+		led_power_orange: power_orange {
+			label = "orange:power";
+			gpios = <&qcom_pinmux 31 GPIO_ACTIVE_HIGH>;
+		};
+		
+		led_active_white: active_white {
+			label = "white:active";
+			gpios = <&qcom_pinmux 32 GPIO_ACTIVE_HIGH>;
+		};
+	};
+
+	i2c-gpio0 {
+		compatible = "i2c-gpio";
+		pinctrl-0 = <&i2c2_pins>;
+		pinctrl-names = "default";
+		
+		sda-gpios = <&qcom_pinmux 29
+			     (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+		scl-gpios = <&qcom_pinmux 30
+			     (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+		i2c-gpio,delay-us = <10>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		tlc591xx@40 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			compatible = "ti,tlc59108";
+			reg = <0x40>;
+
+			red@0 {
+				label = "red:user";
+				reg = <0x0>;
+			};
+
+			green@1 {
+				label = "green:user";
+				reg = <0x1>;
+			};
+
+			blue@2 {
+				label = "blue:user";
+				reg = <0x2>;
+			};
+		};
+	};
+};
+
+&qcom_pinmux {
+	i2c1_pins: i2c1_pins {
+		mux {
+			pins = "gpio24", "gpio25";
+			function = "gsbi2";
+			drive-strength = <2>;
+			bias-pull-up;
+			input;
+		};
+	};
+
+	button_pins: button_pins {
+		mux {
+			pins = "gpio26";
+			function = "gpio";
+			drive-strength = <2>;
+			bias-pull-up;
+		};
+	};
+
+	i2c2_pins: i2c2_pins {
+		mux {
+			pins = "gpio29", "gpio30";
+			function = "gpio";
+			drive-strength = <2>;
+			bias-pull-up;
+			input;
+		};
+	};
+
+	led_pins: led_pins {
+		mux {
+			pins = "gpio31", "gpio32";
+			function = "gpio";
+			drive-strength = <12>;
+			bias-pull-down;
+			output-low;
+		};
+	};
+};
+
+&gsbi2 {
+	qcom,mode = <GSBI_PROT_I2C>;
+	status = "okay";
+};
+
+&gsbi2_i2c {
+	status = "okay";
+
+	pinctrl-0 = <&i2c1_pins>;
+	pinctrl-names = "default";
+	
+	ina2xx@40 {
+		compatible = "ina219";
+		shunt-resistor = <40000>;
+		reg = <0x40>;
+	};
+
+	at24@56 {
+		compatible = "atmel,24c64";
+		pagesize = <32>;
+		reg = <0x56>;
+	};
+};
+
+
+&soc {
+	mdio_gpio0: mdio {
+		compatible = "virtual,mdio-gpio";
```

Comment:
Please try to use ipq8064-mdio driver.
92da53b12dce
0d8098548ecb
e42aca06ab13

Reply:
If `ipq8064-mdio` doesn't works, I think it is ok to use `mdio-gpio`. Thank you for clarification and testing.

Reply:
I revisited this issue after discovering that within `mdio0_pins_active`, function had been defined `gpio`. After correcting this to `mdio`, I was able to get this working without the bitbang.

Reply:
Returning to this issue, it seems 5.4 is still affected but not 5.10. Two options here; revert back to mdio-gpio, or; stick with the ipq8064-mdio driver and patch back the bit-bang into 5.4, thus no further changes needed when 5.4 support is removed.